Doctrine/ORM/Listener/UpdateDocumentListener.php

```diff
@@ -26,9 +26,10 @@ public function __construct(Solr $solr)
     public function postUpdate(LifecycleEventArgs $args)
     {
         $entity = $args->getEntity();
+        $entityChangeSet = $args->getEntityManager()->getUnitOfWork()->getEntityChangeSet($entity);
 
         try {
-            $this->solr->updateDocument($entity);
+            $this->solr->updateDocument($entity, array_keys($entityChangeSet));
```

Comment:
I would not change the signature of public method if we can do the stuff in the class. It would be hide complexity and won't break other projects which depend on this method. `addToIndex` would be a good place for this.